### PR TITLE
Create a driver file for Kamsrtrup Multical 303

### PR DESCRIPTION
**Pull request by thecem**
I created a driver file based on meter_multical603 and enriched it with all known information provided in the telegram.
Based on the manual information ´X30-10-101: C1, Drive by, Standard Registers´

where could i find the driver for "enercal 24/36"

pls review.
thx a lot!

### CODE REVIEW

**Comment by thecem** [comment]
For testing:

https://wmbusmeters.org/analyze/5e442d2c8182788240047af3005025a90c7eB357dd477fdf96495358deB74d67c02ddad4c97B94f2535eea82de2f8317f5efe7e2df9a0d58Bc630d817cdfdB83e4e32060720637cc8577ecd83BB53eBB13f77821d13fdf8165a1fB51681f46:auto:75EDE0CBBB6E126764898645AA366568

**Comment by thecem** [comment]
Data sheet, page 90:

https://koce1-kamstrup.ocecdn.oraclecloud.com/content/published/api/v1.1/assets/CONT37BE438014F34F26BC3AF0C9CC1D0712/native/55122701_E1_GB.pdf?channelToken=ed241bbb18f444908a8fc9ed97ca5d5b

**Comment by thecem** [comment]
After compiling and testing following:

Why is the meter found as an enercal?

```
ea@Solaris-Air:~/GIT/wmbusmeters$ wmbusmeters --analyze=75EDE0CBBB6E126764898645AA366568 5E442D2C8182788240047ABC00502
51677BDFB93B06C53E7452C54F020554687FFEF602F7A221FB3B09BD06DF0F7378D8738A4F2F793B044DD67C54B115EAD562FA4618C13A98FDC32FDA
16A8FC13ACD3605EA5689BA675A2E27C90D0C2F48
Auto driver  : not found!
Best driver  : enercal 24/36
Using driver : enercal 24/36
000   : 5e length (94 bytes)(OK)
001   : 44 dll-c (from meter SND_NR)
002   : 2d2c dll-mfct (KAM)
004   : 81827882 dll-id (82788281)
008   : 40 dll-version
009   : 04 dll-type (Heat meter)
010   : 7a tpl-ci-field (EN 13757-3 Application Layer (short tplh))
011   : bc tpl-acc-field
012   : 00 tpl-sts-field (OK)
013   : 5025 tpl-cfg 2550 (synchronous AES_CBC_IV nb=5 cntn=0 ra=0 hc=0 )
015   : 2f2f decrypt check bytes (OK)
017   : 04 dif (32 Bit Integer/Binary Instantaneous value)
018   : 06 vif (Energy kWh)
019 C!: 00000000 ("total_kwh":0)
023   : 04 dif (32 Bit Integer/Binary Instantaneous value)
024   : 14 vif (Volume 10⁻² m³)
025 C!: EE000000 ("total_m3":2.38)
029   : 04 dif (32 Bit Integer/Binary Instantaneous value)
030   : FF vif (Manufacturer specific)
031   : 07 combinable vif (Reserved)
032 C?: 3D000000
036   : 04 dif (32 Bit Integer/Binary Instantaneous value)
037   : FF vif (Manufacturer specific)
038   : 08 combinable vif (Reserved)
039 C?: 3D000000
043   : 02 dif (16 Bit Integer/Binary Instantaneous value)
044   : 59 vif (Flow temperature 10⁻² °C)
045 C!: 340A ("forward_c":26.12)
047   : 02 dif (16 Bit Integer/Binary Instantaneous value)
048   : 5D vif (Return temperature 10⁻² °C)
049 C!: 400A ("return_c":26.24)
051   : 02 dif (16 Bit Integer/Binary Instantaneous value)
052   : 3B vif (Volume flow l/h)
053 C!: 0000 ("flow_m3h":0)
055   : 02 dif (16 Bit Integer/Binary Instantaneous value)
056   : FF vif (Manufacturer specific)
057   : 22 combinable vif (PerHour)
058 C?: 0000
060   : 02 dif (16 Bit Integer/Binary Instantaneous value)
061   : 6C vif (Date type G)
062 C?: D228
064   : 44 dif (32 Bit Integer/Binary Instantaneous value storagenr=1)
065   : 06 vif (Energy kWh)
066 C!: 00000000 ("target_kwh":0)
070   : 44 dif (32 Bit Integer/Binary Instantaneous value storagenr=1)
071   : 14 vif (Volume 10⁻² m³)
072 C!: 00000000 ("target_m3":0)
076   : 42 dif (16 Bit Integer/Binary Instantaneous value storagenr=1)
077   : 6C vif (Date type G)
078 C!: C128 ("target_date":"2022-08-01")
080   : 2F skip
081   : 2F skip
082   : 2F skip
083   : 2F skip
084   : 2F skip
085   : 2F skip
086   : 2F skip
087   : 2F skip
088   : 2F skip
089   : 2F skip
090   : 2F skip
091   : 2F skip
092   : 2F skip
093   : 2F skip
094   : 2F skip

{
    "media":"heat",
    "meter":"enercal",
    "name":"",
    "id":"82788281",
    "status":"OK",
    "total_kwh":0,
    "target_kwh":0,
    "flow_m3h":0,
    "forward_c":26.12,
    "return_c":26.24,
    "total_m3":2.38,
    "target_m3":0,
    "subunit1_m3":null,
    "subunit1_target_m3":null,
    "target_date":"2022-08-01",
    "timestamp":"2022-08-18T12:59:42Z"
}
```

File: src/meter_multical303.cc [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+    return has_t2_temperature_;
+}
+
+double MeterMultical303::volumeFlow(Unit u)
+{
+    assertQuantity(u, Quantity::Flow);
+    return convert(volume_flow_m3h_, Unit::M3H, u);
+}
+
+double MeterMultical302::targetEnergyConsumption(Unit u)
```
Comment (thecem) [inline src/meter_multical303.cc:179]: Typo 302 should be 303

**Comment by weetmuts** [comment]
Can we close this pull request?

**Comment by thecem** [comment]
Closed since old, „meter“ driver format.

**Pull request closed without merging**